networkdb/delegate.go

```diff
@@ -71,19 +72,33 @@ func (nDB *NetworkDB) handleNodeEvent(nEvent *NodeEvent) bool {
 	nDB.purgeSameNode(n)
 	n.ltime = nEvent.LTime
 
+	var op opType
+	broadcast := false
+
 	switch nEvent.Type {
 	case NodeEventTypeJoin:
 		nDB.Lock()
 		nDB.nodes[n.Name] = n
 		nDB.Unlock()
-		return true
+		op = opCreate
+		broadcast = true
 	case NodeEventTypeLeave:
```

Comment:
I don't think this provides the Leave events similar to pre swarm mode (< 1.12 versions). `NodeEventTypeLeave` is the NetworkDB event that is propagated when a node intentionally leaves swarm (`docker swarm leave`). But in the case of a network partition this will not generate any event. In that case we only populate the `failedNodes` and keep trying to connect again. But in the pre swarm mode a network partition would have resulted in discovery TTL timeout and triggered a Leave ? 

I think we should use the memberlist Join/Lave notification for this purpose, event_delegate.go/NotifyLeave()